task_types.go

```diff
@@ -170,6 +170,40 @@ func SetTaskInlineInputData(data string) TaskIngestionSpecOptions {
 	}
 }
 
+// SetTaskDruidInputSource configures druid reindex input source for the task based ingestion.
+func SetTaskDruidInputSource(datasource string, startTime time.Time, endTime time.Time) TaskIngestionSpecOptions {
+	return func(spec *TaskIngestionSpec) {
+		spec.Spec.IOConfig.InputSource = &InputSource{
+			Type:       "druid",
+			Datasource: datasource,
+			Interval: fmt.Sprintf(
+				"%s/%s",
+				startTime.Format("2006-01-02T15:04:05"),
```

Comment:
nit: what about extracting the format to the const? if we name it `iso8601` or similarly that would be clearer that the `%s/%s` is that in interval in that format.

WDYT?

Reply:
that makes perfect sense, addressed.